src/block_cache/block_chunkcache.c

```diff
@@ -261,6 +261,37 @@ __chunkcache_eviction_thread(void *arg)
  * __wt_chunkcache_get --
  *     If the cache has the data at the given size and offset, copy it into the supplied buffer.
  *     Otherwise, read and cache the chunks containing the requested data.
+ *
+ * The locks protect two data structures: (1) the buckets of the hash table that keeps track of the
+ *     allocated chunks and the (2) LRU list of chunks, used by eviction. Each bucket in the hash
+ *     table is protected by a single spinlock. The LRU list is also protected by a single lock.
+ *
+ * Lock ordering: * If we ever need to hold a bucket lock and the LRU list lock simultaneously, we
+ *     always acquire the bucket lock first and the LRU lock second. * In the current
+ *     implementation, there is never a need to hold two bucket locks at the same time.
+ *
+ * The LRU list lock protects the list of chunks ordered by access recency. We acquire it when we
+ *     insert a newly allocated chunk, remove the evicted chunk, or when we move the accessed chunk
+ *     from the end of the list to the front.
+ *
+ * Bucket locks: A bucket lock protects the chunks that hashed to the same bucket. These chunks are
+ *     held in a linked list, also referred to as the chunk chain in the source code. On a more
+ *     abstract level, the bucket locks protect the information of whether a chunk representing a
+ *     part of the object is present in the cache. Therefore, we must hold the bucket lock whenever
+ *     we are looking for and inserting a new chunk. We must hold the lock throughout the entire
+ *     operation: realizing that the chunk is not present, deciding to cache it, allocating the
+ *     chunks metadata and inserting it into the chain. If we release the lock during this process,
+ *     another thread might cache the same chunk -- bad idea. We insert the new chunk into
+ *     the cache in the invalid state. Once we insert the chunk, we can release the lock. As long as
+ *     the chunk is marked invalid, no other thread will try to re-cache it or to read from it. As a
+ *     result, we can read data from the remote storage into this chunk without holding the lock and
```

Comment:
I'd remove "and this is what the current code does".  I think that's always assumed.